plugins/scaffolder/src/components/ScaffolderPage/ScaffolderPage.tsx

```diff
@@ -91,9 +91,19 @@ export const ScaffolderPage: React.FC<{}> = () => {
           </SupportButton>
         </ContentHeader>
         {!templates && isValidating && <Progress />}
+        {templates && !templates.length && (
+          <Typography variant="body2">
+            Shoot! Looks like you don't have any templates. Check out the
+            documentation{' '}
+            <Link href="docs/backstage/features/software-templates/adding-templates">
+              here!
+            </Link>
+          </Typography>
+        )}
         <Grid container>
           {templates &&
-            templates.map(template => {
+            templates?.length > 0 &&
+            templates!.map(template => {
```

Comment:
`templates?.map` should be enough in this case, no?

Reply:
On the other hand, at this point we know that templates is defined, so the exclamation mark would only be there because the compiler isn't smart enough to know it's unnecessary

Reply:
Nope @shmidt-i, because `templates` is defined, but it's `[]`. So we need to check that there is actually templates returned in either case. Didn't need the `templates!` as it infers it.